### ISSUE TITLE
No migrations are created for mapping defined indexes

### ISSUE THREAD

**Issue by robertoschwald** [grails-plugins/grails-database-migration#74]
When I define indexes in domain mappings, dbm-gorm-diff does not create the migrations.

Environment:

- Grails 3.0.16 
- db-migration 2.0.0.RC4  
- Oracle Java SDK 1.8.0_65
- OSX
- H2, MySQL or PostgreSQL datasource (does not matter, same problem)

Example:
```groovy

class ApiAccessToken {
  String id
  String apiToken
  String deviceId 

  static constraints = {
    apiToken unique:['deviceId']
  }

  static mapping = {
    cache true
    apiToken index: 'apitoken_Idx'
    deviceId index: 'apitoken_Idx'
  }
}
```

For the added apitoken_Idx index, dbm-gorm-diff states:
```
databaseChangeLog = {
}
```

Therefore I created the index migrations manually and applied them:
```groovy
databaseChangeLog = {
     // Note: Added this migration manually, as dbm currently does not create them in Grails 3
    createIndex(indexName: "apitoken_Idx", tableName: "api_access_token_eo") {
      column(name: "api_token_da")
      column(name: "device_id_da")
    }
}
```

BUT: As soon as I run the next dbm-gorm-diff, the plugin creates a migration which drop the indexes created by my manual migration and immediately recreates them:
```groovy
databaseChangeLog = {

    changeSet(author: "roos (generated)", id: "1464877658372-1") {
        dropIndex(indexName: "apitoken_Idx", tableName: "api_access_token_eo")

        createIndex(indexName: "apitoken_Idx", tableName: "api_access_token_eo") {
            column(name: "api_token_da")
            column(name: "device_id_da")
        }
    }
```

When I drop the index in the db and re-run dbm-gorm-diff, it again states: no changes.

**Comment by robertoschwald** [comment]
Sample application see https://github.com/robertoschwald/grails-database-migration-issue74-example

**Comment by robertoschwald** [comment]
When using Grails 2.4.4 with dbm plugin 1.4.0 using the same sample domain class, the indexes are created correctly:
```groovy
databaseChangeLog = {
	changeSet(author: "roos (generated)", id: "1465201783236-1") {
		createTable(tableName: "api_access_token") {
			column(autoIncrement: "true", name: "id", type: "bigint") {
				constraints(nullable: "false", primaryKey: "true", primaryKeyName: "api_access_toPK")
			}

			column(name: "version", type: "bigint") {
				constraints(nullable: "false")
			}

			column(name: "api_token", type: "varchar(255)") {
				constraints(nullable: "false")
			}

			column(name: "device_id", type: "varchar(255)") {
				constraints(nullable: "false")
			}
		}
	}

	changeSet(author: "roos (generated)", id: "1465201783236-2") {
		createIndex(indexName: "apitoken_Idx", tableName: "api_access_token") {
			column(name: "api_token")
			column(name: "device_id")
		}
	}

	changeSet(author: "roos (generated)", id: "1465201783236-3") {
		createIndex(indexName: "unique_api_token", tableName: "api_access_token", unique: "true") {
			column(name: "device_id")
			column(name: "api_token")
		}
	}
}
```

**Comment by robertoschwald** [comment]
Liquibase seems to detect the apitoken_Idx index correctly. diffResult in GroovyDiffToChangeLogCommand contains the index, but the serializer does not seem to generate the changelog entry:

```
INFO liquibase-hibernate - Found table api_access_token
INFO liquibase-hibernate - Found unique constraint null on api_access_token(device_id, api_token)
WARN liquibase-hibernate - Hibernate primary key name is probably truncated. api_access_toPK
WARN liquibase-hibernate - Changing hibernate primary key name to api_access_tokenPK
INFO liquibase-hibernate - Found primary key api_access_tokenPK
INFO liquibase-hibernate - Found index apitoken_Idx
INFO liquibase-hibernate - Found column id bigint
INFO liquibase-hibernate - Found column version bigint
INFO liquibase-hibernate - Found column api_token varchar(255)
INFO liquibase-hibernate - Found column device_id varchar(255)
INFO liquibase-hibernate - Found index apitoken_Idx
```

So seems to be a Liquibase 3.5 problem. See https://github.com/liquibase/liquibase-hibernate/issues/100

**Comment by robertoschwald** [comment]
As stated in the liquibase issue I referenced, HibernateDatabase is blacklisted in some migration types of newer Liquibase versions. Therefore, normal indexes are currently not supported when using plugin version 2.0.0.x. I contacted the author.

**Comment by jameskleeh** [comment]
@robertoschwald This should be fixed in version `3.0.0.BUILD-SNAPSHOT` of the plugin that supports hibernate 5. Can you confirm that?

**Comment by jameskleeh** [comment]
I verified the index diff is now created correctly:

```
databaseChangeLog = {

    changeSet(author: "jameskleeh (generated)", id: "1477666371113-1") {
        createIndex(indexName: "apitoken_Idx", tableName: "person") {
            column(name: "api_token")

            column(name: "device_id")
        }
    }
}
```
